File: project-service/src/main/java/com/kapitonau/projectservice/model/audit/BaseAuditableModelListener.java
package com.kapitonau.projectservice.model.audit;

import jakarta.persistence.PrePersist;
import jakarta.persistence.PreUpdate;
import lombok.extern.slf4j.Slf4j;

import java.time.ZonedDateTime;

@Slf4j
public class BaseAuditableModelListener {


    @PrePersist
    public void prePersist(BaseAuditableModel model) {
        model.setCreatedBy(0L);
        model.setCreatedTime(ZonedDateTime.now());
        model.setModifiedBy(0L);
        model.setModifiedTime(ZonedDateTime.now());
    }

    @PreUpdate
    public void preUpdate(BaseAuditableModel model) {
        model.setModifiedBy(0L);
        model.setModifiedTime(ZonedDateTime.now());
    }

}

File: task-service/src/main/kotlin/com/kapitonau/taskservice/model/audit/BaseAuditableModelListener.kt
package com.kapitonau.taskservice.model.audit

import com.kapitonau.taskservice.model.audit.BaseAuditableModel
import jakarta.persistence.PrePersist
import jakarta.persistence.PreUpdate
import lombok.extern.slf4j.Slf4j
import java.time.ZonedDateTime

@Slf4j
class BaseAuditableModelListener {
    @PrePersist
    fun prePersist(model: BaseAuditableModel) {
        model.createdBy = 0L
        model.createdTime = ZonedDateTime.now()
        model.modifiedBy = 0L
        model.modifiedTime = ZonedDateTime.now()
    }

    @PreUpdate
    fun preUpdate(model: BaseAuditableModel) {
        model.modifiedBy = 0L
        model.modifiedTime = ZonedDateTime.now()
    }
}
